blead/patches/006.patch

```diff
@@ -289,3 +214,155 @@ index 0000000..e55d64f
 +void __updateccsid(int fd, const char* path, int oflag, int perm);
 +
 +#endif
+diff --git a/os390/os390_init.c b/os390/os390_init.c
+new file mode 100644
+index 0000000..e4eb2c2
+--- /dev/null
++++ b/os390/os390_init.c
+@@ -0,0 +1,146 @@
++#include "os390/os390.h"
++#include <string.h>
++#include <sys/stat.h>
++#include <stdio.h>
++#include <stdlib.h>
++#include <sys/ps.h>
++#include <unistd.h>
++#include <stdarg.h>
++#include <varargs.h>
++#include <limits.h>
++#include <_Nascii.h>
++#include <fcntl.h>
++#include <libgen.h>
++
++void init_stdio_streams() {
++    // If stdout is not a TTY and is untagged, tag it as 1047
++    if (0 == __getfdccsid(fileno(stdout))) {
++        struct file_tag tag;
++        tag.ft_ccsid = 1047;
++        tag.ft_txtflag = 1;
++        fcntl(fileno(stdout), F_SETTAG, &tag);
++
++        struct f_cnvrt req = {SETCVTON, 0, 1047};
++        fcntl(fileno(stdout), F_CONTROL_CVT, &req);
++    }
++
++    // If stderr is not a TTY is untagged, tag as 1047
++    if (0 == __getfdccsid(fileno(stderr))) {
++        struct file_tag tag;
++        tag.ft_ccsid = 1047;
++        tag.ft_txtflag = 1;
++        fcntl(fileno(stderr), F_SETTAG, &tag);
++
++        struct f_cnvrt req = {SETCVTON, 0, 1047};
++        fcntl(fileno(stderr), F_CONTROL_CVT, &req);
++    }
++
++    // If stdin not a TTY is untagged, tag as 1047
++    if (0 == __getfdccsid(fileno(stdin))) {
++        struct file_tag tag;
++        tag.ft_ccsid = 1047;
++        tag.ft_txtflag = 1;
++        fcntl(fileno(stdin), F_SETTAG, &tag);
++
++        // Below is needed for /bin/sh STDIN
++        struct f_cnvrt req = {SETCVTON, 0, 1047};
++        fcntl(fileno(stdin), F_CONTROL_CVT, &req);
++    }
++}
++
++int orig_mode;
```

Comment:
should the scope of these be 'static' or does it need to be cross-compilation unit?

Reply:
yep, we can do static here